staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go

```diff
@@ -388,6 +388,12 @@ func (f *ConfigFlags) WithDiscoveryBurst(discoveryBurst int) *ConfigFlags {
 	return f
 }
 
+// WithWrapConfigFn allows providing a wrapper function for the client Config.
+func (f *ConfigFlags) WithWrapConfigFn(wrapConfigFn func(*rest.Config) *rest.Config) *ConfigFlags {
+	f.WrapConfigFn = wrapConfigFn
+	return f
+}
```

Comment:
Do we want to allow chaining of these functions? This is really what I meant when I said this should be a slice.

```
func main() {
	flags := genericclioptions.NewConfigFlags(false)
	flags = flags.WithWrapConfigFn(func(config *rest.Config) *rest.Config {
		fmt.Println("wrap 1")
		return config
	})
	flags = flags.WithWrapConfigFn(func(config *rest.Config) *rest.Config {
		fmt.Println("wrap 2")
		return config
	})
	flags.ToRESTConfig()
}

// wrap 2
// wrap 1
// Unsure of the order we'd want here.
```

Or am I missing something here?

```suggestion
func (f *ConfigFlags) WithWrapConfigFn(wrapConfigFn func(*rest.Config) *rest.Config) *ConfigFlags {
	if f.WrapConfigFn == nil {
		f.WrapConfigFn = wrapConfigFn
	} else {
		old := f.WrapConfigFn
		f.WrapConfigFn = func(config *rest.Config) *rest.Config {
			return old(wrapConfigFn(config))
		}
	}
	return f
}
```

Reply:
Since `WrapConfigFn` both accepts and returns `*rest.Config` you can chain then together on your own, and pass that into `WithWrapConfigFn` that's why I don't think we need or want to make this a slice.